proposals/lip-0045.md

```diff
@@ -1575,6 +1446,47 @@ genesisInteroperabilityStoreSchema = {
 
 Here, the `...` notation, borrowed from [JavaScript ES6 data destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring), indicates that the corresponding schema should be inserted in place, and it is just used for notational convenience.
 
+#### Genesis State Initialization
+
+During the genesis state initialization stage of a genesis block `g`, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the genesis block assets for the Interoperability module and let `interoperabilityAsset = decode(genesisInteroperabilityStoreSchema, genesisBlockAssetBytes)`. Let `chainInfos = interoperabilityAsset.chainInfos`.
+
+##### Genesis Asset Verification
+
+* Within `chainInfos`, check that all entries have a different `chainID`.
+* For every `chainInfo` in `chainInfos`, properties `chainInfo.chainData`, `chainInfo.channelData` and `chainInfo.chainValidators` must be non-empty.
+* For every `chainInfo` in `chainInfos` with non-empty `chainInfo.terminatedOutbox`, the field `chainInfo.terminatedState` must also be non-empty.
```

Comment:
I think that also here this is not correct. What does 'empty' mean precisely in this context? I think you mean 'set to empty bytes', but this is true only if we have intermediate serialization, which is not what we do in the schema. The most elegant solution would be to have non-required `terminatedState` and `terminatedOutbox`, but this is not allowed by Lisk codec. Maybe we need intermediate serialization? Or maybe we can get away with 'empty', but we need to be very precise here

Reply:
I split into several properties to avoid this problem